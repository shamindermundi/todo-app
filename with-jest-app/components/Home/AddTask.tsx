import React, { useState } from "react";

import styles from "./home.module.css";

/**
 * Component to add new task
 */

interface Props {
    setTaskList: Function;
}

const AddTask = ({ setTaskList }: Props) => {
    const [newTask, setNewTask] = useState("");

    // Function to add task in localstorage
    const handleAdd = () => {
        if (newTask.trim() === "") return;

        let taskList: any = localStorage?.getItem("taskList");
        let newList;
        if (taskList) {
            taskList = JSON.parse(taskList);
            newList = [...taskList, newTask];
        } else {
            newList = [newTask];
        }

        setTaskList(newList);
        localStorage.setItem("taskList", JSON.stringify(newList));
        setNewTask("");
    };

    return (
        <div className={styles.addTask}>
            <input
                placeholder="Add your task"
                className={styles.input}
                value={newTask}
                onChange={(e) => setNewTask(e.target.value)}
            />
            <button onClick={() => handleAdd()} className={styles.addButton}>
                Add
            </button>
        </div>
    );
};

export default AddTask;
