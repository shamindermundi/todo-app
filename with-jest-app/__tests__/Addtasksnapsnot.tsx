import { render } from "@testing-library/react";
import AddTask from "@/components/Home/AddTask";

it("renders homepage unchanged", () => {
    const { container } = render(<AddTask setTaskList={() => console.log("first")} />);
    expect(container).toMatchSnapshot();
});
