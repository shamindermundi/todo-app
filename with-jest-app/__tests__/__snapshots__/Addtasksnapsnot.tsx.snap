// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <form
    class="addTask"
  >
    <input
      class="input"
      placeholder="Add your task"
      value=""
    />
    <button
      class="addButton"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
